src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -337,267 +235,241 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
+                if (dependencyGroups.Any())
+                {
+                    metadata.Dependencies ??= new List<Dependency>();
 
-                        foreach (PackageDependencyGroup dependencyGroup in dependencyGroups)
-                        {
-                            dependencyGroup.Packages.ToList().ForEach((dependency) => metadata.Dependencies.Add(new Dependency() { Package = dependency.ToString(), Framework = dependencyGroup.TargetFramework.ToString()}));
-                        }
+                    foreach (PackageDependencyGroup dependencyGroup in dependencyGroups)
+                    {
+                        dependencyGroup.Packages.ToList().ForEach((dependency) => metadata.Dependencies.Add(new Dependency() { Package = dependency.ToString(), Framework = dependencyGroup.TargetFramework?.ToString()}));
                     }
+                }
 
-                    // keywords
-                    metadata.Keywords = OssUtilities.ConvertJSONToList(OssUtilities.GetJSONPropertyIfExists(versionElement, "tags"));
+                // keywords
+                metadata.Keywords = new List<string>(packageVersion.Tags.Split(", "));
 
-                    // licenses
+                // licenses
+                if (packageVersion.LicenseMetadata is not null)
+                {
                     metadata.Licenses ??= new List<License>();
                     metadata.Licenses.Add(new License()
                     {
-                        Name = OssUtilities.GetJSONPropertyStringIfExists(versionElement, "licenseExpression"),
-                        Url = OssUtilities.GetJSONPropertyStringIfExists(versionElement, "licenseUrl")
+                        Name = packageVersion.LicenseMetadata.License,
+                        Url = packageVersion.LicenseMetadata.LicenseUrl.ToString()
                     });
-                    
-                    // publishing info
-                    metadata.UploadTime = OssUtilities.GetJSONPropertyStringIfExists(versionElement, "published");
                 }
-            }
 
-            if (latestVersion is not null)
-            {
-                metadata.LatestPackageVersion = latestVersion.ToString();
+                // publishing info
+                metadata.UploadTime = packageVersion.Published?.ToString("MM/dd/yy HH:mm:ss zz");
             }
 
             return metadata;
         }
 
-        public JsonElement GetLatestCatalogEntry(JsonElement root)
-        {
-            return root.GetProperty("items").EnumerateArray().Last() // Last CatalogPage
-                .GetProperty("items").EnumerateArray().Last() // Last Package
-                .GetProperty("catalogEntry"); // Get the entry for the most recent package version
-        }
-        
-        public override JsonElement? GetVersionElement(JsonDocument? contentJSON, Version desiredVersion)
+        private async Task<JsonElement?> GetCatalogEntry(JsonDocument doc, PackageURL purl)
```

Comment:
Do this method and the ones below have any purpose anymore?

Reply:
No, removed.